app/components/CodeEditor.tsx

```diff
@@ -96,6 +96,21 @@ export function CodeEditor(opts: CodeEditorProps) {
     }
   }, [selection, view, setSelectionRef.current]);
 
+  useEffect(() => {
+    const listener = (event: WindowEventMap["keydown"]) => {
+      if ((event.ctrlKey || event.metaKey) && event.key === "a") {
+        event.preventDefault();
+        view?.dispatch({ selection: { anchor: 0, head: state?.doc.length } });
+      }
+    };
+
+    window.addEventListener("keydown", listener);
+
+    return () => {
+      window.removeEventListener("keydown", listener);
+    };
+  }, [view, state]);
```

Comment:
I would recommend `tinykeys` for shortcuts management for this kind of project.

```js
import tinykeys from 'tinykeys'

function useHotKey(params: Parameters<typeof tinykeys>[1]) {
  useEffect(() => {
    const unsubscribe = tinykeys(window, params)
    return () => {
      unsubscribe()
    }
  }, [ params ])
}
```

```js
useHotKey({
    "$mod+A": () => {
        event.preventDefault();
        view?.dispatch({ selection: { anchor: 0, head: state?.doc.length } });
    }
  })
```

Mac: $mod = Meta (⌘)
Windows/Linux: $mod = Control

Reply:
We actually already use the `react-hotkeys-hook` package in a bunch of different places, e.g.: https://github.com/triggerdotdev/jsonhero-web/blob/main/app/components/SearchBar.tsx#L18

Reply:
Should I change this to `react-hotkeys-hook`?

Reply:
Yes please 👍